### PR TITLE
Add webhook endpoint to enable on-demand ISR

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Enable [on-demand ISR](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#using-on-demand-revalidation). This endpoint is supposed to be used as a webhook that I've setup in Storyblok and will run every time someone publishes new content.

Add new env var REVALIDATE_SECRET to secure endpoint

## Justify why they are needed

This lets us rebuild individual pages when content changes instead of having to deploy the whole site or wait for the cache to invalidate.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

[GRW-1319]: https://hedvig.atlassian.net/browse/GRW-1319?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #357** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/357" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/357?utm_source=stack-comment).

**Review by guilhermespopolin — APPROVED** [review]
That's cool. Didn't know about this on-demand ISR thing :)

File: apps/store/src/pages/api/revalidate.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const SLUG_TO_ROUTE_MAP: Record<string, string> = {
+  home: '/',
+}
+
+const STORYBLOK_API = new StoryblokClient({
+  accessToken: process.env.NEXT_PUBLIC_STORYBLOK_ACCESS_TOKEN,
+})
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  // check for secret to confirm this is a valid request
```
Comment (alebedev) [inline apps/store/src/pages/api/revalidate.ts:21]: Can we remove this comment?  I think code itself is just as clear. Maybe error message can be more specific to make it even more readable

Comment (robinandeer) [reply apps/store/src/pages/api/revalidate.ts:21]: aha yes good point, I will remove it

**Pull request merged**